backend/cfg/cfg_prologue.ml

```diff
@@ -153,69 +153,132 @@ module Reachable_epilogues = struct
   let from_block (t : t) (label : Label.t) = Label.Tbl.find t label
 end
 
-let can_place_prologue (prologue_label : Label.t) (cfg : Cfg.t)
+let descendants (cfg : Cfg.t) (block : Cfg.basic_block) : Label.Set.t =
+  let visited = ref Label.Set.empty in
+  let rec collect label =
+    if not (Label.Set.mem label !visited)
+    then (
+      visited := Label.Set.add label !visited;
+      let block = Cfg.get_block_exn cfg label in
+      Label.Set.iter
+        (fun succ_label -> collect succ_label)
+        (Cfg.successor_labels ~normal:true ~exn:true block))
+  in
+  collect block.start;
+  !visited
```

Comment:
(I wonder whether it is generic
enough to be moved to `Cfg`;
not specific to this PR so a
`CR-soon` is maybe better.)